doc/examples/custom_type.rst

```diff
@@ -0,0 +1,146 @@
+Custom Type Example
+===================
+
+This is an example of using a custom type with PyMongo. The example here shows
+how to subclass :class:`~bson.codec_options.TypeCodecBase` to write a type
+codec, which is used to populate a :class:`~bson.codec_options.TypeRegistry`.
+The type registry can then be used to create a custom-type-aware
+:class:`~pymongo.collection.Collection`. Read and write operations
+issued against the resulting collection object transparently manipulate
+documents as they are saved or retrieved from MongoDB.
+
+
+Setup
+-----
+
+We’ll start by getting a clean database to use for the example:
+
+.. doctest::
+
+  >>> from pymongo import MongoClient
+  >>> client = MongoClient()
+  >>> client.drop_database('custom_type_example')
+  >>> db = client.custom_type_example
+
+
+Since the purpose of the example is to demonstrate working with custom types,
+we’ll need a custom data type to use. For this example, we will be working with
+the :py:class:`~decimal.Decimal` type from Python's standard library. Since the
```

Comment:
Does `:py:class:` work for you? It doesn't generate a link for me (running Sphinx v1.6.5).

Reply:
I needed to add the intersphinx inventory. It works now, but I am not sure whether to link to Python 2 or 3 docs. Changes introduced in `61f9ede` make it so that we link to the standard library docs corresponding to the python version being used to run `sphinx` on to compile PyMongo docs.

Reply:
I think we should always link to Python 3.